bundles/product-api-schedule-price-import/src/FondOfKudu/Zed/ProductApiSchedulePriceImport/Business/Model/SalePriceProductConcreteHandler.php

```diff
@@ -0,0 +1,157 @@
+                );
+
+            if ($priceProductScheduleTransfer === null) {
+                $this->create($priceProductTransfer, $productConcreteTransfer);
+            } else {
+                $this->update($priceProductScheduleTransfer, $productConcreteTransfer);
+            }
+        }
+
+        return $productConcreteTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PriceProductTransfer $priceProductTransfer
+     * @param \Generated\Shared\Transfer\ProductConcreteTransfer $productConcreteTransfer
+     *
+     * @return void
+     */
+    protected function create(
+        PriceProductTransfer $priceProductTransfer,
+        ProductConcreteTransfer $productConcreteTransfer
+    ): void {
+        $priceProductScheduleTransfer = $this->priceProductScheduleMapper->createFromProductConcreteTransfer(
+            $priceProductTransfer,
+            $productConcreteTransfer,
+        );
+
+        $this->priceProductScheduleFacade->createAndApplyPriceProductSchedule($priceProductScheduleTransfer);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PriceProductScheduleTransfer $priceProductScheduleTransfer
+     * @param \Generated\Shared\Transfer\ProductConcreteTransfer $productConcreteTransfer
+     *
+     * @return void
+     */
+    protected function update(
+        PriceProductScheduleTransfer $priceProductScheduleTransfer,
+        ProductConcreteTransfer $productConcreteTransfer
+    ): void {
+        $productConcreteAttributes = $productConcreteTransfer->getAttributes();
+        $specialPriceFrom = $productConcreteAttributes[$this->apiSchedulePriceImportConfig->getProductAttributeSalePriceFrom()];
+        $specialPriceTo = $productConcreteAttributes[$this->apiSchedulePriceImportConfig->getProductAttributeSalePriceTo()];
+        $specialPrice = $productConcreteAttributes[$this->apiSchedulePriceImportConfig->getProductAttributeSalePrice()];
+
+        $priceProductScheduleTransfer
+            ->setActiveFrom($specialPriceFrom)
+            ->setActiveTo($specialPriceTo)
+            ->getPriceProduct()->getMoneyValue()->setGrossAmount($specialPrice);
+
+        $this->priceProductScheduleFacade->updateAndApplyPriceProductSchedule($priceProductScheduleTransfer);
+    }
+
+    /**
+     * @param array $productAttributes
+     *
+     * @return bool
+     */
+    protected function validateSpecialPriceAttributes(array $productAttributes): bool
+    {
+        $required = [
+            $this->apiSchedulePriceImportConfig->getProductAttributeSalePrice(),
+            $this->apiSchedulePriceImportConfig->getProductAttributeSalePriceFrom(),
+            $this->apiSchedulePriceImportConfig->getProductAttributeSalePriceTo(),
+        ];
+
+        foreach ($required as $item) {
+            if (!isset($productAttributes[$item])) {
+                return false;
+            }
+
+            if (!$productAttributes[$item]) {
+                return false;
+            }
```

Comment:
won't isset already check if the variable is true? because it will check if its not null? if we need a specific check I would check for the required type

Reply:
No, isset() checks if an array index exists. Even if the value of the index is false or null, isset() returns true. Ideally, the indexes should always be transmitted through the middleware, just for safety.

Reply:
Did you have something specific in mind? I actually find it quite nice as it makes the different paths clear. You probably weren't just thinking of something like that, right?

```
if ($priceProductScheduleTransfer === null) {
   $this->create($priceProductTransfer, $productConcreteTransfer);

   continue;
}

$this->update($priceProductScheduleTransfer, $productConcreteTransfer);`
```

Reply:
Yes.. but then we should change the second check right? Shoudln't this check against string ? Or even date